sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java

```diff
@@ -62,4 +64,47 @@ public AnalyzedStatement analyze(ShowSchemas node, Analysis analysis) {
         return selectStatementAnalyzer.visitQuery((Query) statement, analysis);
     }
 
+    public AnalyzedStatement analyze(ShowTables node, Analysis analysis) {
+
+
+        /**
+         * <code>
+         *     SHOW TABLES [{ FROM | IN } schema_name ] [ { LIKE 'pattern' | WHERE expr } ];
+         * </code>
+         * needs to be rewritten to select query:
+         * <code>
+         *     SELECT distinct(table_name) as table_name
+         *     FROM information_schema.tables
+         *     [ WHERE ( schema_name = 'schema_name' | schema_name NOT IN ( 'sys', 'information_schema' ) )
+         *      [ AND ( table_name LIKE 'pattern' | <where-clause > ) ]
+         *     ]
+         *     ORDER BY 1;
+         * </code>
+         */
+        StringBuilder sb = new StringBuilder("SELECT distinct(table_name) as table_name FROM information_schema.tables ");
```

Comment:
consider creating an AstVisitor that creates String statements from Show... nodes.
Like this you can easily unittest this stuff and the parsing and select-analyzing can be refactored to a single method.